providers/provider_data.go

```diff
@@ -21,3 +25,17 @@ type ProviderData struct {
 
 // Data returns the ProviderData
 func (p *ProviderData) Data() *ProviderData { return p }
+
+func (p *ProviderData) GetClientSecret() (ClientSecret string, err error) {
+	if p.ClientSecret != "" || p.ClientSecretFile == "" {
+		return p.ClientSecret, nil
+	}
+
+	// Getting ClientSecret can fail in runtime so we need to report it without returning the file name to the user
+	fileClientSecret, err := ioutil.ReadFile(p.ClientSecretFile)
+	if err != nil {
+		logger.Printf("error reading client secret file %s: %s", p.ClientSecretFile, err)
+		return "", errors.New("could not read client secret file")
+	}
+	return string(FileClientSecret), nil
```

Comment:
This one needs changing as well, currently this doesn't build
```suggestion
	return string(fileClientSecret), nil
```

Reply:
Thanks!